storage/innobase/srv/srv0start.cc

```diff
@@ -776,6 +776,22 @@ static dberr_t srv_undo_tablespace_fixup_num(space_id_t space_num) {
   if (!undo::is_active_truncate_log_present(space_num)) {
     return (DB_SUCCESS);
   }
+#ifdef XTRABACKUP
+  if (srv_backup_mode) {
+    if (opt_lock_ddl) {
+      xb::warn() << "Found _trunc.log in server dirctory. ";
+      return (DB_SUCCESS);
+    } else {
+      xb::error()
+          << "Found _trunc.log file in server directory. Undo truncation might "
+             "be concurrently running, Try backup after undo truncation is "
+             "complete. These _trunc.log could be left over of previous undo "
+             "truncation remove them safely to continue backup. You can also "
+             "run backup with --lock-ddl=true which blocks concurrent undo "
+             "truncation";
```

Comment:
arent we supposed to return with error from here?

Reply:
fixed...
Missed that during development. lock test was successful

> 2022-07-12T13:01:08.348170+05:30 0 [ERROR] [MY-011825] [Xtrabackup] Found _trunc.log file in server directory. Undo truncation might be concurrently running, Try backup after undo truncation is complete. These _trunc.log could be left over of previous undo truncation remove them safely to continue backup. You can also run backup with --lock-ddl=true which blocks concurrent undo truncation
> 2022-07-12T13:01:08.348192+05:30 0 [Note] [MY-012902] [InnoDB] Undo tablespace number 1 was being truncated when mysqld quit.
> 2022-07-12T13:01:08.348201+05:30 0 [ERROR] [MY-012903] [InnoDB] Cannot recover a truncated undo tablespace in read-only mode
> 2022-07-12T13:01:08.348230+05:30 0 [ERROR] [MY-011825] [Xtrabackup] xb_load_tablespaces() failed with error code 57

and after  returning with DB_ERROR

> > 2022-07-12T13:02:18.704277+05:30 0 [ERROR] [MY-011825] [Xtrabackup] Found _trunc.log file in server directory. Undo truncation might be concurrently running, Try backup after undo truncation is complete. These _trunc.log could be left over of previous undo truncation remove them safely to continue backup. You can also run backup with --lock-ddl=true which blocks concurrent undo truncation
> 2022-07-12T13:02:18.704341+05:30 0 [ERROR] [MY-011825] [Xtrabackup] xb_load_tablespaces() failed with error code 11